src/Components/Browser.JS/src/Services/UriHelper.ts

```diff
@@ -65,8 +65,17 @@ export function navigateTo(uri: string, forceLoad: boolean) {
   const absoluteUri = toAbsoluteUri(uri);
 
   if (!forceLoad && isWithinBaseUriSpace(absoluteUri)) {
+    // It's an internal URL, so do client-side navigation
     performInternalNavigation(absoluteUri, false);
+  } else if (forceLoad && location.href === uri) {
+    // Force-loading the same URL you're already on requires special handling to avoid
+    // triggering browser-specific behavior issues.
+    // For details about what this fixes and why, see https://github.com/aspnet/AspNetCore/pull/10839
+    const temporaryUri = uri + '?';
+    history.replaceState(null, '', temporaryUri);
```

Comment:
Nice. Does this run in to the same issue if you hit go back + forward + back?

Reply:
In my experiments, that sequence of actions worked fine after this fix.

The fix causes the temporary client-side history entry to get replaced with a regular server-side one, so all back-or-forward sequences cause a full page load when you go across that barrier.

Reply:
Note that I've only been trying this on desktop browsers, and only in the specific case where it's about navigation to an external URL.

I don't *think* it's likely to cause regressions in any other case, since the new code path only applies when you're doing a force-load to the same URL you're already on, which is extremely unlikely to be something you're doing in any other case/